backend/src/Model/TreeNode.ts

```diff
@@ -17,6 +19,7 @@ export class TreeNode<ViewModel extends Destroyable> {
   public onMessage = new EventDispatcher<Message>()
   public onDestroy = new EventDispatcher<TreeNode<ViewModel>>()
   public isTree = false
+  public type: TopicDataType = 'json'
```

Comment:
@thomasnordquist i'm a bit confused about node.type here? Should't `type` be derived from the selected `Decoder`?